### PR TITLE
Added new parameter to allow excluding project files from response.

### PR DESCRIPTION
**Pull request by jtbm37**
Hi

This PR relates to https://github.com/OmniSharp/omnisharp-server/pull/202
We need to exclude the project files from the response as they are not used by the clients.

### CODE REVIEW

File: src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs [hunk 1]
```diff
@@ -24,11 +25,15 @@ public class ProjectSystemController
 
         [HttpPost("/projects")]
         [HttpGet("/projects")]
-        public WorkspaceInformationResponse ProjectInformation()
+        public WorkspaceInformationResponse ProjectInformation(ProjectInformationRequest request)
         {
+            var includeSourceFiles = true;
```
Comment (nosami) [inline src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs]: It would be better if this default value was set on the request

File: src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs [hunk 2]
```diff
@@ -24,11 +25,15 @@ public class ProjectSystemController
...
+            if (request != null)
```
Comment (nosami) [inline src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Don't need to test for the request being null

File: src/OmniSharp/Models/v1/MsBuildWorkspaceInformation.cs [hunk 3]
```diff
@@ -6,11 +6,20 @@ namespace OmniSharp.Models
...
         {
             SolutionPath = msbuildContext.SolutionPath;
 
-            Projects = msbuildContext.Projects.Values.Select(p => new MSBuildProject(p));
+            Projects = msbuildContext
+                .Projects
+                .Values
+                .OrderBy(x => x.AssemblyName)
+                .Select(p => {
+                    var prj = new MSBuildProject(p);
```
Comment (nosami) [inline src/OmniSharp/Models/v1/MsBuildWorkspaceInformation.cs]: rename prj to project

File: src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs (hunk 2 above)
Comment (jtbm37) [inline src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs]: If clients are already making use of this action method and they do not send any parameters it will throw an NRE (NullReferenceException). Are you ok with that ?

File: src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs (hunk 1 above)
Comment (jtbm37) [inline src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs]: I will rename it to `excludeSourceFiles` as well. I wanted to preserve the default behaviour which was to include the source files.

File: src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs (hunk 2 above)
Comment (nosami) [inline src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Yeah. We don't check anywhere else.

**Comment by nosami** [comment]
Ah, sorry... I just realised that ProjectInformationRequest is a new parameter, so the request will need a null check. I'll add it back in.

File: src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs [hunk 4]
```diff
@@ -24,11 +25,11 @@ public class ProjectSystemController
...
         [HttpPost("/projects")]
         [HttpGet("/projects")]
-        public WorkspaceInformationResponse ProjectInformation()
+        public WorkspaceInformationResponse ProjectInformation(ProjectInformationRequest request)
         {
             return new WorkspaceInformationResponse
             {
-                MSBuild = new MsBuildWorkspaceInformation(_msbuildContext),
+                MSBuild = new MsBuildWorkspaceInformation(_msbuildContext, request.ExcludeSourceFiles),
                 Dnx = new DnxWorkspaceInformation(_dnxContext),
```
Comment (jrieken) [inline src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Why only for MSBuild projects?
